File: python/kindergarten-garden/kindergarten_garden.py
class Garden:
    def __init__(self, diagram, students=None):
        self.diagram = diagram

        if students is not None:
            self.students = sorted(students)
        else:
            self.students = [
                "Alice",
                "Bob",
                "Charlie",
                "David",
                "Eve",
                "Fret",
                "Ginny",
                "Harriet",
                "Ileana",
                "Joseph",
                "Kincaid",
                "Larry",
            ]

        self.plant_map = {
            "G": "Grass",
            "V": "Violets",
            "C": "Clover",
            "R": "Radishes"
        }

    def plants(self, name):
        diagram_arr = self.diagram.split("\n")
        index = self.students.index(name) * 2

        letters = diagram_arr[0][index: index + 2] + diagram_arr[1][index: index + 2]

        return [self.plant_map[letter] for letter in letters]

File: java/kindergarten-garden/src/main/java/KindergartenGarden.java
import java.util.Arrays;
import java.util.Collection;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.stream.Collectors;

class KindergartenGarden {

    private final String garden;
    private final Map<String, Integer> studentsMap = new HashMap<>() {{
        put("Alice", 1);
        put("Bob", 2);
        put("Charlie", 3);
        put("David", 4);
        put("Eve", 5);
        put("Fred", 6);
        put("Ginny", 7);
        put("Harriet", 8);
        put("Ileana", 9);
        put("Joseph", 10);
        put("Kincaid", 11);
        put("Larry", 12);
    }};

    KindergartenGarden(String garden) {
        this.garden = garden;
    }

    List<Plant> getPlantsOfStudent(String student) {
        return Arrays.stream(garden.split("\n"))
                .map(row -> row.chars()
                        .skip((studentsMap.get(student) - 1) * 2L)
                        .limit(2)
                        .mapToObj(plantChar -> Plant.getPlant((char) plantChar))
                        .collect(Collectors.toList())
                )
                .flatMap(Collection::stream)
                .collect(Collectors.toList());
    }

}
